spec/lib/whitehall_importer/create_edition_spec.rb

```diff
@@ -86,38 +86,148 @@
         )
+      end
 
-        edition = described_class.call(document_import: document_import,
-                                       whitehall_edition: whitehall_edition,
-                                       user_ids: user_ids)
+      it "has a draft revision" do
+        timeline_entry = edition.timeline_entries.first
+        expect(timeline_entry.attributes).to match a_hash_including(
+          "entry_type" => "whitehall_migration",
+          "created_at" => draft_revision["created_at"],
+          "created_by_id" => first_user.id,
+        )
+        expect(timeline_entry.details.attributes).to match a_hash_including(
+          "entry_type" => "first_created",
+          "contents" => "{}",
+        )
+      end
 
-        expect(edition.timeline_entries.first).to be_whitehall_migration
-        expect(edition.timeline_entries.first.details).to be_first_created
-        expect(edition.timeline_entries.first.created_at).to eq(2.days.ago)
-        expect(edition.timeline_entries.first.created_by_id).to eq(first_user.id)
+      it "has a published revision" do
+        timeline_entry = edition.timeline_entries.second
+        expect(timeline_entry.attributes).to match a_hash_including(
+          "entry_type" => "whitehall_migration",
+          "created_at" => published_revision["created_at"],
+          "created_by_id" => second_user.id,
+        )
+        expect(timeline_entry.details.attributes).to match a_hash_including(
+          "entry_type" => "published",
+          "contents" => "{}",
+        )
+      end
 
-        expect(edition.timeline_entries.last).to be_whitehall_migration
-        expect(edition.timeline_entries.last.details).to be_published
-        expect(edition.timeline_entries.last.created_at).to eq(1.day.ago)
-        expect(edition.timeline_entries.last.created_by_id).to eq(second_user.id)
+      it "has an editorial remark" do
+        timeline_entry = edition.timeline_entries.third
+        expect(timeline_entry.attributes).to match a_hash_including(
+          "entry_type" => "whitehall_migration",
+          "created_at" => editorial_remark["created_at"],
+          "created_by_id" => first_user.id,
+        )
+        expect(timeline_entry.details.attributes).to match a_hash_including(
+          "entry_type" => "internal_note",
+          "contents" => "{\"contents\":\"Another note\"}",
+        )
+      end
+
+      it "has a fact check request" do
+        timeline_entry = edition.timeline_entries.fourth
+        expect(timeline_entry.attributes).to match a_hash_including(
+          "entry_type" => "whitehall_migration",
+          "created_at" => fact_check["created_at"],
+          "created_by_id" => second_user.id,
+        )
+        expect(timeline_entry.details.attributes).to match a_hash_including(
+          "entry_type" => "fact_check_request",
+          "contents" => "{\"email_address\":\"someone@somewhere.com\",\"instructions\":\"Do something\"}",
+        )
```

Comment:
I think something has gone wrong for this to be a string. I'd expect this to be a hash.

Reply:
Hopefully, this is now resolved!  Thanks - nice spot.